### PR TITLE
removing as:Link in favour of as:Object

### PR DESCRIPTION
**Pull request by elf-pavlik**
As proposal for [ISSUE-14: as:Link adds a lot of complexity, if we keep it we need to clarify consequences of using it instead of as:Object](https://www.w3.org/Social/track/issues/14) I started removing as:Link from examples and for encoding of images define new **ImageBlob** class and **encoding** property.

I will draw a diagram explaining problem with as:Link and add it to this description.

Not ready to merge, work in progress to discuss issue around concrete examples from the spec!

### CODE REVIEW

**Comment by elf-pavlik** [comment]
Illustrating some of the main issues with *as:Link*
1. instance of *as:Link* becomes subject of *as:mediaType* not the instance of *as:Image* (or *as:ImageBlob*).
2. when querying for all the resources related by *as:image* we get bunch of **[blank nodes](http://www.w3.org/TR/rdf11-concepts/#section-blank-nodes)** with *as:href* relation instead of actual objects we would expect.

as:Object vs. as:Link diagram

**Comment by bblfish** [comment]
I spent a lot of time thinking about Link relations 10 years ago when working on the Atom 
XML syntax ( [RFC 4287]( https://tools.ietf.org/html/rfc4287) ), which the Web Linking [RFC 5988](https://tools.ietf.org/html/rfc5988) refers to .  Since Activity Streams is meant to be compatible with Atom syntax, it may be interesting to see how else one can model Link relations.

In developing the [Atom Owl ontology](http://bblfish.net/work/atom-owl/2006-06-06/AtomOwl.html#Link)  I tried to make sure I could use it to map Atom  to RDF and back, whilst also producing as clean as possible RDF ( for which there is an isomorphic mapping to json-ld ).

> Link Class - see §4.2.7 of rfc 4287 spec. Note that the href, hreflang and lengtlength attributes of rfc 4287 have been moved to the content class and a content relation from the Link to a Content class has been made. The href is translated onto the :src relation of the Content class. A link is a reified relation between an entry an a content, for the purpose of adding some metadata such as a title. If it is a reified statement, it is a special type of such a statement, one that implies always the reified statement itself (see N3 rule) 

So you’d have an Atom Entry with a Link relation like the following ( in Turtle )

```turtle
:entry1 :link [ a :Link;
                :rel iana:alternate ;
                :to [ :src <http://example.org/2003/12/13/atom03>; ]
              ] .
```

Since not everyone reads turtle, we can describe it in terms of a graph like this

linkrelation-extended

So we can already see the difference with the current activity streams modelling, ( as I understand it by reading Elf’s comments above ) is that the relation from the entry to the  object is the link relation not the iana:alternate relation . Elf’s example is even clearer. Instead of iana:alternate he uses as:image . Doing this makes it immediately  clear that it is odd to think of as:image as relating a person to something as abstract as a link relation as Robert Sanderson pointed out.

What is interesting is that if you look at the above diagram and twist your mind a bit you can make the following mapping betwen the relations used above and the [rdf Reification vocabulary]( http://www.w3.org/TR/rdf-schema/#ch_reificationvocab)

linkrelation-extended-map

namely:

* the inverse of awol:link is a subrelation of rdf:subject
* awol:rel is a subrelation of rdf:predicate
* awol:object is a subrelation of rdf:object

If you add to that a reasonable assumption that anyone making a link relation statement in an atom xml file is asserting the truth of that Reification then if follows from the above that

```turtle
<entry> iana:alternate <document> .
```

which is the relation drawn in red in the second diagram.

Ie we can directly link the entry to the document via the iana:alternate relation or the as:image relation.
Both are two ways to say the same thing. The re-ificiation just gives you a way to add metadata on the
link relation itself. This could be useful if say you thought that the title was the title of the link, rather than the title of the object of the relation. But I think that over the past 8 years it seems to have become clear that the title was not the title of the reified link, but the title of the object of the link. (see RFC5988 on Web Linking §5.4 ).  As that has become clear it means also that there is no need for the re-ification anymore ( which I only added to Atom-Owl out of prudence). Elf is clearly correct in that the direct relation is a lot simpler than the re-ified one.

So I agree with Elf and Robert Sanderson that this part of AS2.0 could be modelled much more simply.
Given that I’d propose that AS2 link directly  to the document as shown above.

**Comment by jasnell** [comment]
Issue resolved on todays call: maintain status quo and keep as:Link

**Comment by gobengo** [comment]
@jasnell for those of us not on the call today, can you summarize that conclusion? @elf-pavlik and @bblfish make convincing arguments (though I'm admittedly not in a position to form an obvious conclusion).

I can always wait for minutes on the wiki like [those posted for the 4/21 call](http://www.w3.org/wiki/Socialwg/2015-04-21-minutes)
